### PR TITLE
Update Spark Plugin to v1beta2

### PR DESCRIPTION
**Pull request by akhurana001**
Move spark plugin to  v1beta2 multi-version of Operator. Ref: https://github.com/lyft/flyte/issues/573

### CODE REVIEW

File: go/tasks/plugins/k8s/spark/spark.go [hunk 1]
```diff
@@ -17,7 +17,7 @@ import (
 	"github.com/lyft/flyteplugins/go/tasks/pluginmachinery/utils"
 	"k8s.io/client-go/kubernetes/scheme"
 
-	sparkOp "github.com/GoogleCloudPlatform/spark-on-k8s-operator/pkg/apis/sparkoperator.k8s.io/v1beta1"
+	sparkOp "github.com/GoogleCloudPlatform/spark-on-k8s-operator/pkg/apis/sparkoperator.k8s.io/v1beta2"
```
Comment (kumare3) [inline go/tasks/plugins/k8s/spark/spark.go:20]: thats it no changes in the API?

**Comment by kumare3** [comment]
When you do this, can you also change the oss flyte deployment to deploy the new version?

**Review by kumare3 — APPROVED** [review]

File: go/tasks/plugins/k8s/spark/spark.go (hunk 1 above)
Comment (akhurana001) [reply go/tasks/plugins/k8s/spark/spark.go:20]: There are some change like types of some fields in the Spec has changed but we don't use that in flyteplugins i.e. only set those config via spark_conf. The  one other change is see is that status is now a sub-resource and the operator needs additional perms for the same.

**Comment by akhurana001** [comment]
> When you do this, can you also change the oss flyte deployment to deploy the new version?

Will do in https://github.com/lyft/flyte/issues/573 once I create a new propeller release .

**Pull request merged**